io/key/key.go

```diff
@@ -64,6 +65,23 @@ type EditEvent struct {
 	Text string
 }
 
+// InputMode changes the on-screen-keyboard type. That hints the
+// type of data that might be entered by the user.
+type InputMode uint8
```

Comment:
I think your "hint" is a better name than "mode", not least because the underlying platform may well ignore it. So how about `InputHint`?